src/MCPClient/lib/clientScripts/dip_generation_helper.py

```diff
@@ -0,0 +1,132 @@
+            for row in reader:
+                filename = row[0]
+                ref_id = row[1]
+                file_info[filename] = ref_id
+    return file_info
+
+def parse_archivesspace_ids(sip_path, sip_uuid):
+    """
+    Parse an archivesspaceids.csv to pre-populate the matching GUI.
+
+    :param sip_path: Path to the SIP to check for an archivesspaceids.csv
+    :param sip_uuid: UUID of the SIP to auto-populate ArchivesSpace IDs for
+    :return: 0 on success, 1 on failure
+    """
+    # Check for archivesspaceids.csv
+    csv_paths = archivematicaFunctions.find_metadata_files(sip_path, 'archivesspaceids.csv')
+    if not csv_paths:
+        print('No archivesspaceids.csv files found, exiting')
+        return 0
+
+    file_info = parse_archivesspaceids_csv(csv_paths)
+    if not file_info:
+        print('No information found in archivesspaceids.csv files')
+        return 1
+    print(file_info)
+
+    # Create client
+    client = create_archivesspace_client()
+    if not client:
+        return 1
+
+    for filename, ref_id in file_info.items():
+        # Get file object (for fileUUID, to see if in DIP)
+        print(filename, ref_id, '%SIPLocation%' + filename)
+        try:
+
+            f = models.File.objects.get(
+                Q(originallocation='%transferDirectory%' + filename) |
+                Q(originallocation='%transferDirectory%objects/' + filename) |
+                Q(originallocation='%SIPDirectory%' + filename) |
+                Q(originallocation='%SIPDirectory%objects/' + filename),
+                sip_id=sip_uuid
+            )
+        except models.File.DoesNotExist:
+            print(filename, 'not found in database, skipping')
+            continue
+        except models.File.MultipleObjectsReturned:
+            print('Multiple entries for', filename, 'found in database, skipping')
+            continue
+        print('File:', f)
+
+        # Query ref_id to client for resource_id
+        resource = client.find_by_field('identifier', ref_id)
+        try:
+            resource_id = resource[0]['id']
+        except IndexError:
+            print('ArchivesSpace did not return an ID for', ref_id)
+            print('Returned', resource)
+            continue
+        print('Resource ID:', resource_id)
+
+        # Add to ArchivesSpaceDIPObjectResourcePairing
+        models.ArchivesSpaceDIPObjectResourcePairing.objects.create(
+            dipuuid=sip_uuid,
+            fileuuid=f.uuid,
+            resourceid=resource_id,
+        )
+
+    # Check if any files were processed?
+    return 0
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Parse metadata for DIP helpers')
+    parser.add_argument('--sipUUID', required=True, help='%SIPUUID%')
+    parser.add_argument('--sipPath', required=True, help='%SIPDirectory%')
+    args = parser.parse_args()
+
+    rc = 0
+
+    rc = rc or parse_archivesspace_ids(args.sipPath, args.sipUUID)
```

Comment:
I was thinking this script would be used for multiple DIP helpers, and the script should return failure if any one of them failed, so the RC should be the or'd results of all the helpers. Does this seem reasonable?